### DISCUSSION TITLE
Vertical Lines not being separated in the faceted plot

### DISCUSSION THREAD

**Discussion by lolitanaz314** [vega/altair#3223] [General]
I have this sample data:
```
import altair as alt
import pandas as pd

# Create a sample DataFrame
data = pd.DataFrame({
    'media': ['display', 'gdn', 'gdn', 'gdn', 'gdn'],
    'x': [1, 2, 3, 4, 5],
    'y': [10, 20, 15, 30, 25],
    'vert_line_x': [2, 5, 5, 5, 5],
    'vert_line_y1': [0, 0, 0, 0, 0],
    'vert_line_y2': [15, 20, 20, 20, 20],
    })

# Create a chart with points
base_chart = alt.Chart(data).mark_point().encode(
    x='x:O',
    y='y:Q'
)

# Create a vertical line that goes up to y=20
line_chart = alt.Chart(data).mark_rule(
    strokeDash=[5, 5]).encode(
      x='vert_line_x:O',
      x2='vert_line_x:O',
      y='vert_line_y1:Q',
      y2='vert_line_y2:Q'
    )

# Combine the chart and the line using layer
final_chart = alt.layer(base_chart, line_chart).facet('media', data=data)

# Show the chart
final_chart
```
I want each faceted plot to show one line (one for display and one for gdn). I thought I separated it by media channel by specifying "facet" but there's two lines per plot. how would I separate them?

<img width="226" alt="Screenshot 2023-10-10 at 11 17 41 AM" src="https://github.com/altair-viz/altair/assets/29680448/259ef84d-6b6e-4e98-8b82-2b3039f1cdaf">

Thanks!

**Comment by joelostblom** [comment]
You don't need to specify the `data` both in the individual charts and the facet. If you put it in just one of them it should work (but note that you are creating multiple lines in the gdn facet)

**Reply by lolitanaz314** [reply]
thank you, removing data=data from the facet worked!
